dll/win32/msafd/misc/dllmain.c

```diff
@@ -1974,18 +2021,28 @@ WSPConnect(SOCKET Handle,
     ConnectInfo->UseSAN = FALSE;
     ConnectInfo->Unknown = 0;
 
-    /* FIXME: Handle Async Connect */
     if (Socket->SharedData->NonBlocking)
     {
-        ERR("Async Connect UNIMPLEMENTED!\n");
+        APCFunction = &AfdConnectAPC; // should be a private io completition function inside us
+        APCContext = HeapAlloc(GlobalHeap, 0, sizeof(AFDCONNECTAPCCONTEXT));
```

Comment:
```suggestion
        APCContext = HeapAlloc(GlobalHeap, 0, sizeof(*APCContext));
```

Reply:
I'd like to keep the original style for consistency.

Reply:
Maybe, but using something more correct should be privileged (Here for example, so that if one changes the type of the APCContext in the future, that the code stays correct without having to fix all the instances of the type here and there)

Reply:
OK.